### PR TITLE
fix(mastodon): align Threads integration with Meta API docs

### PR DESCRIPTION
**Pull request by alphatownsman**
## Summary

Verified the Threads (threads.net) integration against Meta's official Threads API documentation. The OAuth, token and publishing flows were confirmed correct; this PR fixes the deviations found:

- **Replies now work**: request `threads_manage_replies` scope and send `reply_to_id` when posting. The old TODO ("waiting for Meta to confirm it's bug or permission issue") was a missing permission, not a Meta bug — per the [Create Replies doc](https://developers.facebook.com/docs/threads/retrieve-and-manage-replies/create-replies/), reply creation requires that scope. Tokens authorized before the scope was added fall back to a top-level post.
- **Text truncation**: Threads text posts max out at 500 characters; long content previously failed container creation with a generic error. Text is now truncated to 480 chars (buffer for emoji counted by UTF-8 byte length), preserving the item link if truncation would cut it off.
- **Token refresh**: long-lived tokens under 24 hours old cannot be refreshed per the [Long-Lived Tokens doc](https://developers.facebook.com/docs/threads/get-started/long-lived-tokens/); `check_alive` now treats an unexpired token as alive when refresh is rejected, instead of marking freshly-authenticated accounts unreachable.
- **Meta callbacks**: `threads/uninstall` (deauthorize) and `threads/delete` (data deletion request) now handle Meta's POSTed `signed_request` per spec — HMAC-SHA256 verification, token cleared on deauthorize, ThreadsAccount data removed on deletion request with the required `{url, confirmation_code}` JSON response and a status page. The NeoDB user account and their content are kept; GET on these endpoints still redirects to the data page as before.
- **Hygiene**: `get_single` uses GET instead of POST, the `delete` helper wraps `requests.delete` instead of `requests.post`, and OAuth authorize URL parameters are properly urlencoded.

Note for deployment: the `threads_manage_replies` permission must also be enabled for the app in the Meta App Dashboard; existing users need to re-authorize for replies to thread.

## Tests

- 23 new tests in `tests/mastodon/test_threads.py`: truncation shapes (comment/review/note/no-link), auth URL encoding, container+publish flow, reply fallback, `check_alive` token-refresh edge cases, signed_request verification, and callback views (deauthorize, deletion, invalid payload, GET redirect).
- Full suite: 1464 passed.

### CODE REVIEW

File: mastodon/views/threads.py [hunk 1]
```diff
@@ -59,11 +67,78 @@ def threads_oauth(request: HttpRequest):
...
+def _parse_signed_request(signed_request: str) -> dict | None:
+    """parse and verify a signed_request POSTed by Meta (base64url, HMAC-SHA256)"""
+    if "." not in signed_request:
+        return None
+    encoded_sig, payload = signed_request.split(".", 1)
+    try:
+        sig = base64.urlsafe_b64decode(encoded_sig + "=" * (-len(encoded_sig) % 4))
+        data = json.loads(base64.urlsafe_b64decode(payload + "=" * (-len(payload) % 4)))
+    except ValueError, TypeError:
+        return None
```
[the remark this answers is not in the document]
Comment (alphatownsman) [reply mastodon/views/threads.py:80]: This is valid syntax on Python 3.14+ (PEP 758, unparenthesized except). The project targets Python 3.14 and ruff format actively rewrites the parenthesized form to this; CI lint/pytest/CodeQL all pass.

File: mastodon/views/threads.py [hunk 2]
```diff
@@ -59,11 +67,78 @@ def threads_oauth(request: HttpRequest):
...
+    secret = SiteConfig.system.threads_app_secret
+    expected_sig = hmac.new(secret.encode(), payload.encode(), hashlib.sha256).digest()
+    if not secret or not hmac.compare_digest(sig, expected_sig):
+        logger.warning("Threads signed_request signature mismatch")
+        return None
```
[the remark this answers is not in the document]
Comment (alphatownsman) [reply mastodon/views/threads.py:88]: threads_app_secret is typed str with default "" in SiteConfig.SystemOptions and the DB loader filters None values, so AttributeError cannot occur — but moved the empty check before the HMAC computation anyway for clarity.

File: mastodon/views/threads.py [hunk 3]
```diff
@@ -59,11 +67,78 @@ def threads_oauth(request: HttpRequest):
...
+        + "?code="
+        + confirmation_code
+    )
+    return JsonResponse({"url": status_url, "confirmation_code": confirmation_code})
+
+
+@require_http_methods(["GET"])
+def threads_delete_status(request: HttpRequest):
+    """status page for Threads data deletion requests"""
+    code = request.GET.get("code", "")
```
[the remark this answers is not in the document]
Comment (alphatownsman) [reply mastodon/views/threads.py]: Django template autoescaping already prevents XSS here ({{ secondary_msg }} in common/error.html is escaped), but added an alphanumeric sanitize on the code param as defense-in-depth in 87884897.

File: mastodon/models/threads.py [hunk 4]
```diff
@@ -37,23 +38,52 @@
...
+        # some content (e.g. note) embeds the item link in a trailing footer
+        urls = re.findall(r"https?://\S+", text)
+        link = urls[-1] if urls else ""
+    suffix = "……\n" + link if link else "……"
+    head = text[: max(THREADS_MAX_TEXT_LENGTH - len(suffix), 0)]
+    if link and link in head:
+        # link survives truncation, no need to re-append it
+        suffix = "……"
+        head = text[: THREADS_MAX_TEXT_LENGTH - len(suffix)]
+    return head.rstrip() + suffix
```
[the remark this answers is not in the document]
Comment (alphatownsman) [reply mastodon/models/threads.py]: Fixed in 87884897: when the truncation point falls inside the link, the cut now happens right before the link starts. Added a regression test.

File: mastodon/models/threads.py [hunk 5]
```diff
@@ -137,19 +167,28 @@ def get_profile(
...
             return None
-        url = f"https://graph.threads.net/v1.0/{user_id}/threads_publish?creation_id={media_container_id}&access_token={token}"
-        response = post(url)
+        url = f"https://graph.threads.net/v1.0/{user_id}/threads_publish"
+        response = post(
+            url, params={"creation_id": media_container_id, "access_token": token}
+        )
         if response.status_code != 200:
             logger.debug(f"Error {url} {response.status_code} {response.content}")
             return None
```
[the remark this answers is not in the document]
Comment (alphatownsman) [reply mastodon/models/threads.py]: Fixed in 87884897: both post calls now catch network exceptions and return None, consistent with obtain_token/refresh_token/get_profile. Callers already caught these via the generic handler in sync_to_threads, but this keeps the module style uniform and produces the user-facing failure message.
